scripts/1-fetch/doaj_fetch.py

```diff
@@ -0,0 +1,591 @@
+    initialize_data_file(FILE_DOAJ_COUNT, HEADER_COUNT)
+    initialize_data_file(FILE_DOAJ_SUBJECT_REPORT, HEADER_SUBJECT_REPORT)
+    initialize_data_file(FILE_DOAJ_LANGUAGE, HEADER_LANGUAGE)
+    initialize_data_file(FILE_DOAJ_YEAR, HEADER_YEAR)
+    initialize_data_file(FILE_DOAJ_PUBLISHER, HEADER_PUBLISHER)
+
+
+def extract_license_type(license_info):
+    """Extract CC license type from DOAJ license information."""
+    if not license_info:
+        return "UNKNOWN CC legal tool"
+    for lic in license_info:
+        lic_type = lic.get("type", "")
+        if lic_type in CC_LICENSE_TYPES:
+            return lic_type
+    return "UNKNOWN CC legal tool"
+
+
+def process_journals(session, args):
+    """Process DOAJ journals with CC licenses using API v4."""
+    LOGGER.info("Fetching DOAJ journals...")
+
+    license_counts = Counter()
+    subject_counts = defaultdict(Counter)
+    language_counts = defaultdict(Counter)
+    year_counts = defaultdict(Counter)
+    publisher_counts = defaultdict(Counter)
+
+    total_processed = 0
+    page = 1
+    page_size = 100
+
+    while total_processed < args.limit:
+        LOGGER.info(f"Fetching journals page {page}...")
+
+        url = f"{BASE_URL}/journals/*"
+        params = {"pageSize": page_size, "page": page}
+
+        try:
+            response = session.get(url, params=params, timeout=30)
+            response.raise_for_status()
+            data = response.json()
+        except requests.exceptions.RequestException as e:
+            if hasattr(e, "response") and e.response.status_code == 400:
+                LOGGER.info(f"Reached end of available data at page {page}")
+                break
+            else:
+                LOGGER.error(f"Failed to fetch journals page {page}: {e}")
+                raise shared.QuantifyingException(
+                    f"Critical API error on page {page}: {e}", exit_code=1
+                )
+        except (ValueError, KeyError) as e:
+            LOGGER.error(f"Failed to parse JSON response on page {page}: {e}")
+            raise shared.QuantifyingException(
+                f"Critical JSON parsing error on page {page}: {e}", exit_code=1
+            )
+
+        try:
+            results = data.get("results", [])
+            if not results:
+                break
+        except (AttributeError, TypeError) as e:
+            LOGGER.error(f"Invalid API response structure on page {page}: {e}")
+            raise shared.QuantifyingException(
+                f"Critical API response format error on page {page}: {e}",
+                exit_code=1,
+            )
+
+        for journal in results:
+            if total_processed >= args.limit:
+                break
+
+            try:
+                bibjson = journal.get("bibjson", {})
+
+                # Check for CC license
+                license_info = bibjson.get("license")
+                if not license_info:
+                    continue
```

Comment:
I was wondering, wouldn't it be better to log a warning here saying that you skipped this journal because there is no CC license?